src/piece/AbstractPiece.java

```diff
@@ -2,7 +2,7 @@
 
 import java.awt.Image;
 
-public abstract class AbstractPiece {
+public abstract class AbstractPiece implements Cloneable{
```

Comment:
Ended up reverting this change, turned out there was no reason for pieces to be cloned.